pkg/controller/direct/bigquery/table_controller.go

```diff
@@ -0,0 +1,351 @@
+	mapCtx := &direct.MapContext{}
+
+	if err := a.normalizeReferences(ctx); err != nil {
+		return fmt.Errorf("normalizing reference for update: %w", err)
+	}
+
+	desired := a.desired.DeepCopy()
+	table := BigQueryTableSpec_ToProto(mapCtx, &desired.Spec)
+
+	// For table created from external data configuration,
+	// If no schema was provided, the Update request would fail.
+	// Getting schema from actual object.
+	if table.ExternalDataConfiguration != nil && table.Schema == nil && table.ExternalDataConfiguration.Schema == nil {
+		table.Schema = a.actual.Schema
+		for _, field := range table.Schema.Fields {
+			setEmptyPolicyTagsInSchema(field)
+		}
+	}
+	eq, err := TableEq(a.actual, table)
+
+	if err != nil {
+		return err
+	}
+	// No diff detected.
+	if eq {
+		return nil
+	}
+
+	a.customTableLogic(table)
+	parent := a.id.Parent()
+
+	if mapCtx.Err() != nil {
+		return fmt.Errorf("error updating Table %s: %w", a.id.ID(), mapCtx.Err())
+	}
+
+	// TODO: Update annotation logic after confirming requirement.
+	// if a.desired.ObjectMeta.Annotations != nil {
+	// 	unmanaged, ok := a.desired.ObjectMeta.Annotations["cnrm.cloud.google.com/unmanaged"]
+	// 	if ok && unmanaged != "" {
+	// 		unmanagedFields := strings.Split(unmanaged, ",")
+	// 		for _, field := range unmanagedFields {
+	// 			// This ability is only intended for spec.schema field for the moment.
+	// 			if field == "spec.schema" {
+	// 				table.Schema = nil
+	// 			}
+	// 		}
+	// 		// Make PATCH call with nil schema to avoid schema being updated.
+	// 		res, err := a.gcpService.Patch(parent.ProjectID, parent.DatasetID, a.id.ID(), table).Do()
+	// 		if err != nil {
+	// 			return fmt.Errorf("error updating Table %s: %w", a.id.ID(), err)
+	// 		}
+	// 		return a.UpdateStatusForUpdate(ctx, updateOp, res)
+	// 	}
+	// }
+	res, err := a.gcpService.Update(parent.ProjectID, parent.DatasetID, a.id.ID(), table).Do()
+	if err != nil {
+		return fmt.Errorf("error updating Table %s: %w", a.id.ID(), err)
+	}
+	return a.UpdateStatusForUpdate(ctx, updateOp, res)
+}
+
+func (a *Adapter) UpdateStatusForUpdate(ctx context.Context, updateOp *directbase.UpdateOperation, created *bigquery.Table) error {
+	mapCtx := &direct.MapContext{}
+	status := BigQueryTableStatus_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	external := a.id.String()
+	status.ExternalRef = &external
+	if err := updateOp.UpdateStatus(ctx, status, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error) {
+	if a.actual == nil {
+		return nil, fmt.Errorf("Find() not called")
+	}
+	u := &unstructured.Unstructured{}
+
+	obj := &krm.BigQueryTable{}
+	mapCtx := &direct.MapContext{}
+	obj.Spec = direct.ValueOf(BigQueryTableSpec_FromProto(mapCtx, a.actual))
+	if mapCtx.Err() != nil {
+		return nil, mapCtx.Err()
+	}
+	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
```

Comment:
You'll also want to handle identity related fields, i.e. `datasetRef` before converting obj to unstructured.